enterprise/backend/test/metabase_enterprise/serialization/v2/e2e/yaml_test.clj

```diff
@@ -27,244 +25,335 @@
+               :segment                 (many-random-fks 30 {:spec-gen {:definition {:filter [:!= [:field 60 nil] 50],
+                                                                                     :source-table 4}}}
+                                                         {:table_id   [:t 100]
+                                                          :creator_id [:u 10]})
+               :native-query-snippet    (many-random-fks 10 {} {:creator_id    [:u 10]
+                                                                :collection_id [:coll 10 100]})
+               :timeline                (many-random-fks 10 {} {:creator_id    [:u 10]
+                                                                :collection_id [:coll 100]})
+               :timeline-event          (many-random-fks 90 {} {:timeline_id   [:timeline 10]})
+               :pulse                   (vec (concat
+                                               ;; 10 classic pulses, from collections
+                                               (many-random-fks 10 {} {:collection_id [:coll 100]})
+                                               ;; 10 classic pulses, no collection
+                                               (many-random-fks 10 {:refs {:collection_id ::rs/omit}} {})
+                                               ;; 10 dashboard subs
+                                               (many-random-fks 10 {:refs {:collection_id ::rs/omit}}
+                                                                {:dashboard_id  [:d 100]})))
+               :pulse-card              (vec (concat
+                                               ;; 60 pulse cards for the classic pulses
+                                               (many-random-fks 60 {} {:card_id       [:c 100]
+                                                                       :pulse_id      [:pulse 10]})
+                                               ;; 60 pulse cards connected to dashcards for the dashboard subs
+                                               (many-random-fks 60 {} {:card_id           [:c 100]
+                                                                       :pulse_id          [:pulse 10 20]
+                                                                       :dashboard_card_id [:dc 300]})))
+               :pulse-channel           (vec (concat
+                                               ;; 15 channels for the classic pulses
+                                               (many-random-fks 15 {} {:pulse_id  [:pulse 10]})
+                                               ;; 15 channels for the dashboard subs
+                                               (many-random-fks 15 {} {:pulse_id  [:pulse 10 20]})))
+               :pulse-channel-recipient (many-random-fks 40 {} {:pulse_channel_id [:pulse-channel 30]
+                                                                :user_id          [:u 100]})}))
```

Comment:
Since this is not new this PR I'm not suggesting you change it here, but I think having this gigantic wall of code is a little hard to read. Maybe abstracting these out into a `random-entity-specs` multimethod (not sure about the name) that dispatches off of the model keyword would make this a little more readable. Not sure about that. But maybe that would mean it could be used in other places too more easily